frontend/components/CameraView.js

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import { View, TouchableOpacity, ActivityIndicator, StyleSheet } from 'react-native';
+import { Camera, Permissions } from 'expo';
+import { Text, Button } from 'react-native-elements';
+import { Ionicons } from '@expo/vector-icons';
+
+export default class CameraView extends React.Component {
+
+  state = {
+    hasCameraPermission: null,
+    type: Camera.Constants.Type.back,
+  };
+
+  async componentDidMount() {
+    const { status } = await Permissions.askAsync(Permissions.CAMERA);
+    this.setState({ hasCameraPermission: status === 'granted' });
+  }
+
+  handlePictureTaken = (image) => {
+    // When the image data is returned to us then exit the camera view and pass the image back to the main view
+    this.props.navigation.goBack();
+    this.props.navigation.getParam('handlePictureTaken', () => {alert("Reached camera from unknown route")})(image);
+  }
+
+  render() {
+    const { hasCameraPermission } = this.state;
+    // Check that permission was granted
+    if (hasCameraPermission === null) {
+      return <View />;
+    } else if (hasCameraPermission === false) {
+      return <Text>No access to camera</Text>;
+    } else {
+      return (
+        <View style={styles.container}>
```

Comment:
This view is quite lengthy, so it would be useful to have comments throughout that help explain what it is doing.

Reply:
Definitely, I can add some comments